Added notifications and sound alerts in client.js

Roslot opened the pull request:
It goes without saying that this should be tested before deployment, but I mention it because I hardly tested it myself outside of a single browser and OS.

This is a port of what my web extension does.

For deployment in prod, **you will need to add /audio/notify.ogg on your webserver** unless you make edits.

Apologies for the haphazard changes to white space in some parts.

marzavec commented:
Hi! Thanks for the PR, a few people have been asking for this. After testing the code, I found:

FireFox 66.0.5:
     Sound Notification ✔️
     Screen Notification ✔️

Chrome 74.0.3729.131:
     Sound Notification ✔️
     Screen Notification ❌

Edge 42.17134.1.0:
     Sound Notification ✔️ 
     Screen Notification ❌

All three did request and were granted permission. I verified the check boxes were checked before running the test. No errors in any consoles. The bold highlighting is also working in all three.

Roslot commented:
Fixed, view specifics below.

### Edge
It turns out that sound was not working on Edge - the audio file was in an .ogg format, I changed to .mp3 for better compatibility.

_However_, native notifications were working to begin with. You should check the windows Notification Center, you might have Do Not Disturb/Quiet Hours on, but the received notifications come in for me on Windows.

### Chrome
The notifications are working, but the user needs to click/tap at least once on the page before notifications can be sent.
Here is an error you might see with the latest commit:

![https://i.imgur.com/gJC7TQ9.png](https://i.imgur.com/gJC7TQ9.png)

This is similar to how Firefox prevents video sound from auto-playing upon page load, it is normal behavior.

# Defaults
- Sound and native screen notifications are off by default.
- Upon loading the page for the first time, you do not get asked for notification permissions. I think it was intrusive and annoying.
- When you switch on the `Screen notification` option, you will be requested for permission and shown your grant/decline of permission inside the chat only once.
    - If neither Block or Allow is chosen, the user will be asked again when someone @mentions them.

@marzavec If you'd like to change these defaults, let me know.

marzavec commented:
Defaults are good. I am still not getting Chrome screen notifications between three different devices, no errors in console. But if it's only me, I will merge this. There is one issue that I'll tag you in after merging.

MinusGix commented:
I'm getting notifications on Vivaldi (Chromium) on Ubuntu.
